### PR TITLE
Adds amax and amin references

### PR DESCRIPTION
**Pull request by ngimel**
Also extends reference testing to error inputs.

### CODE REVIEW

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 # Experimental module containing prototype Python references for existing
 #   PyTorch operations.
 
-all = [
+__all__ = [
```
Comment (mruberry) [inline torch/_refs/__init__.py:26]: Thank you!

**Review by mruberry — APPROVED** [review]
Wahoo!

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -1045,6 +1046,50 @@ def sum(
         output_dtype_kind=REDUCTION_OUTPUT_TYPE_KIND.SAME,
     )
 
+def amin(
+    a: Tensor,
+    dim: Union[Optional[int], Optional[List[int]]] = None,
+    keepdim: bool = False,
+    *,
+    dtype=None,
```
Comment (Chillee) [inline torch/_refs/__init__.py:1054]: I don't think PyTorch amin has a dtype kwarg, right? Are we going to unify dtype kwargs on all reduction ops?

Comment (ngimel) [reply torch/_refs/__init__.py:1054]: Oh you are right, let me fix the ref.

Comment (Chillee) [reply torch/_refs/__init__.py:1054]: Other than that, looks identical to the ATen operator.

Comment (ngimel) [reply torch/_refs/__init__.py]: That's because for aten amin/amax the overloads are actually sane.

**Pull request closed without merging**